include/sys/emmc.h

```diff
@@ -0,0 +1,303 @@
+#ifndef _SYS_EMMC_H_
+#define _SYS_EMMC_H_
+
+#include <inttypes.h>
+#include <sys/mimiker.h>
+#include <sys/device.h>
+#include <sys/endian.h>
+#include <stddef.h>
+
+/* These operation modes are described in JESD86-A441, with a brief summary
+   at page 27 */
+typedef enum emmc_op_mode {
+  EMMCMODE_BOOT,
+  EMMCMODE_CARD_ID,
+  EMMCMODE_INTR,
+  EMMCMODE_TRANSFER,
+  EMMCMODE_INA,
+} emmc_op_mode_t;
+
+typedef enum emmc_result {
+  EMMC_OK,
+  EMMC_ERR_IERR,
+  EMMC_ERR_TIMEOUT,
+  EMMC_ERR_NOT_RESPONDING,
+  EMMC_ERR_INTR,
+  EMMC_ERR_BUSY,
+} emmc_result_t;
+
+/* Response types R1-R5 are described in JESD86-A441, page 94.
+ * Response R1 and R1b are normally both represented as `EMMCRESP_R1`,
+ * because they are structurally identical. */
+typedef enum emmc_resp_type {
```

Comment:
why is it `emmc_resp_type` and not `emmc_response_type`?
I like full names, but if it has to be short then let all be short/long/full
In line 42. you have full name.

Reply:
You are right, there's an inconsistency. I think I'm going to charge `emmc_response` to `emmc_resp`. Shorter names are preferred as long as given the context one can make out their meaning. Look at _FreeBSD_'s code. There's plenty of short names there.